### PR TITLE
Add support for "#aarrggbb" colors

### PR DESCRIPTION
**Pull request by cookieguru**
As mentioned in #1142

### CODE REVIEW

**Comment by ralfstx** [comment]
Shouldn't it by RGBA (`#rrggbbaa`)?

**Comment by cookieguru** [comment]
@ralfstx From #1142:
> It would be nice to set `#aarrggbb` (prefered) or `#rrggbbaa`

**Comment by mpost** [comment]
My thought on prefering `aarrggbb` over `rrggbbaa` was that, when the alpha component is at the start, the binary value of the rrrggbb portion of the string does not shift, making it easier to extract certain colors because the are always at the same location. After looking around on some other projects and general usage of the definition, RGBA seems to be more common. Therefore i would also suggest to shift the AA to the end.

@cookieguru what do you think about it?

**Comment by cookieguru** [comment]
@mpost I don't see any inherit advantages of one over the other since there is no standard for this notation yet, but would lean towards `#rrggbbaa` since it's closer to the `r, g, b, a` syntax.

I saw this issue come through and since it was easy and can be done all in JS I thought I'd pick it up so you guys have more time to focus on the native code :)

**Comment by mpost** [comment]
@cookieguru We highly appreciate you taking the time to add this feature. If you could update the change to `rgba` than we would be happy accept it.

**Comment by cookieguru** [comment]
@mpost Pushed the updated version.

**Comment by ralfstx** [comment]
@cookieguru Actually, 8 and 4 digit hex colors are part of CSS4 and it's `#rrggbbaa` and `#rgba`: https://www.w3.org/TR/css-color-4/#hex-notation

**Comment by cookieguru** [comment]
@ralfstx Wow, I didn't even know CSS4 was in the works.  Cool stuff!  And even more of a reason to use `#RRGGBBAA`.

**Comment by ralfstx** [comment]
@cookieguru Your change looks good, thank you! I wonder, when we support 3, 6, and 8 digits wouldn't it be consequent to support `#rgba` as well?

**Comment by cookieguru** [comment]
Added shorthand `#rgba`

File: doc/types.md [hunk 1]
```diff
@@ -72,6 +74,8 @@ Examples:
 ```js
 "#f00"
 "#ff0000"
+"#ff00007c" //50% opacity red
```
Comment (ralfstx) [inline doc/types.md]: 50% would be hex `80`

File: doc/types.md [hunk 2]
```diff
@@ -72,6 +74,8 @@ Examples:
...
+"#ff06" //60% opacity yellow
```
Comment (ralfstx) [inline doc/types.md]: `6` should be equivalent to `66`, which represents 40%. Please fix the documentation.
See http://stackoverflow.com/questions/15852122/hex-transparency-in-colors

**Comment by cookieguru** [comment]
@ralfstx Thanks for catching those

**Comment by cookieguru** [comment]
Let's try reopening this and seeing if the commit magically reappears...

**Comment by cookieguru** [comment]
Yay!

**Comment by cookieguru** [comment]
Except I see a bunch more commits than I intended...

**Comment by cookieguru** [comment]
I'll just cherry pick to a new branch.

**Pull request closed without merging**